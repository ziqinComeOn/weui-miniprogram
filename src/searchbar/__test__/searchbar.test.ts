import path from 'path'
import simulate from 'miniprogram-simulate'

describe('searchbar', () => {
    const searchbar = simulate.load(path.resolve(__dirname, '../searchbar'), 'mp-searchbar')

    test('basic', async () => {
        const onSelectResult = jest.fn()

        const wrapper = simulate.render(
            simulate.load({
                usingComponents: {
                    'mp-searchbar': searchbar
                },
                template:
                    '<mp-searchbar id="searchbar" search="{{search}}" bindselectresult="onSelectResult"></mp-searchbar>',
                ready() {
                    this.setData({
                        search: this.search.bind(this)
                    })
                },
                methods: {
                    search(value: string) {
                        return Promise.resolve([{ text: '搜索结果' + value, value: value }])
                    },
                    onSelectResult
                }
            })
        )
        wrapper.attach(document.createElement('parent-wrapper'))
        const comp = wrapper.querySelector('#searchbar')
        await simulate.sleep(0)
        expect(wrapper.toJSON()).toMatchSnapshot()

        const label = comp.querySelector('.weui-search-bar__label')
        const input = comp.querySelector('.weui-search-bar__input')
        label.dispatchEvent('tap')
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        input.dispatchEvent('input', { detail: { value: '123' } })
        await simulate.sleep(1000)
        expect(comp.toJSON()).toMatchSnapshot()

        const results = comp.querySelectorAll('.result')
        results[0].dispatchEvent('tap')
        await simulate.sleep(100)
        expect(comp.toJSON()).toMatchSnapshot()
        expect(onSelectResult).toHaveBeenCalledTimes(1)
        expect(onSelectResult.mock.calls[0][0].detail).toStrictEqual({
            index: 0,
            item: { text: '搜索结果123', value: '123' }
        })

        const cancel = comp.querySelector('.weui-search-bar__cancel-btn')
        cancel.dispatchEvent('tap')
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        label.dispatchEvent('tap')
        await simulate.sleep(0)
        const clear = comp.querySelector('.weui-icon-clear')
        clear.dispatchEvent('tap')
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
    })
})
