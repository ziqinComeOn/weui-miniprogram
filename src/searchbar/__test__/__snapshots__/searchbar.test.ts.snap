// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchbar basic 1`] = `
<main>
  <mp-searchbar
    id="searchbar"
    bind:selectresult="onSelectResult"
  >
    <wx-view
      class="weui-search-bar  "
    >
      <wx-view
        class="weui-search-bar__form"
      >
        <wx-view
          class="weui-search-bar__box"
        >
          <wx-icon
            class="weui-icon-search"
            size="12"
            type="search"
          />
          <wx-input
            class="weui-search-bar__input"
            focus="{{false}}"
            placeholder="搜索"
            type="text"
            value=""
            bind:blur="inputBlur"
            bind:focus="inputFocus"
            bind:input="inputChange"
          />
        </wx-view>
        <wx-label
          class="weui-search-bar__label"
          bind:tap="showInput"
        >
          <wx-icon
            class="weui-icon-search"
            size="12"
            type="search"
          />
          <wx-text
            class="weui-search-bar__text"
          >
            搜索
          </wx-text>
        </wx-label>
      </wx-view>
    </wx-view>
  </mp-searchbar>
</main>
`;

exports[`searchbar basic 2`] = `
<mp-searchbar
  id="searchbar"
  bind:selectresult="onSelectResult"
>
  <wx-view
    class="weui-search-bar weui-search-bar_focusing "
  >
    <wx-view
      class="weui-search-bar__form"
    >
      <wx-view
        class="weui-search-bar__box"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-input
          class="weui-search-bar__input"
          focus="{{true}}"
          placeholder="搜索"
          type="text"
          value=""
          bind:blur="inputBlur"
          bind:focus="inputFocus"
          bind:input="inputChange"
        />
      </wx-view>
      <wx-label
        class="weui-search-bar__label"
        bind:tap="showInput"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-text
          class="weui-search-bar__text"
        >
          搜索
        </wx-text>
      </wx-label>
    </wx-view>
    <wx-view
      class="weui-search-bar__cancel-btn"
      bind:tap="hideInput"
    >
      取消
    </wx-view>
  </wx-view>
</mp-searchbar>
`;

exports[`searchbar basic 3`] = `
<mp-searchbar
  id="searchbar"
  bind:selectresult="onSelectResult"
>
  <wx-view
    class="weui-search-bar weui-search-bar_focusing "
  >
    <wx-view
      class="weui-search-bar__form"
    >
      <wx-view
        class="weui-search-bar__box"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-input
          class="weui-search-bar__input"
          focus="{{true}}"
          placeholder="搜索"
          type="text"
          value="123"
          bind:blur="inputBlur"
          bind:focus="inputFocus"
          bind:input="inputChange"
        />
        <wx-text
          class="weui-icon-clear"
          bind:tap="clearInput"
        />
      </wx-view>
      <wx-label
        class="weui-search-bar__label"
        bind:tap="showInput"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-text
          class="weui-search-bar__text"
        >
          搜索
        </wx-text>
      </wx-label>
    </wx-view>
    <wx-view
      class="weui-search-bar__cancel-btn"
      bind:tap="hideInput"
    >
      取消
    </wx-view>
  </wx-view>
  <mp-cells>
    <wx-view
      class=" searchbar-result  weui-cells__group  "
    >
      <wx-view
        class="weui-cells weui-cells_after-title "
      >
        <mp-cell
          class="result"
          data-index="{{0}}"
          bind:tap="selectResult"
        >
          <wx-view
            class="weui-cell    "
            hoverClass="weui-cell_active"
            bind:tap="navigateTo"
          >
            <wx-view
              class="weui-cell__hd "
            />
            <wx-view
              class="weui-cell__bd weui-cell_primary"
            >
              <wx-view>
                搜索结果123
              </wx-view>
            </wx-view>
            <wx-view
              class="weui-cell__ft "
            />
          </wx-view>
        </mp-cell>
      </wx-view>
    </wx-view>
  </mp-cells>
</mp-searchbar>
`;

exports[`searchbar basic 4`] = `
<mp-searchbar
  id="searchbar"
  bind:selectresult="onSelectResult"
>
  <wx-view
    class="weui-search-bar weui-search-bar_focusing "
  >
    <wx-view
      class="weui-search-bar__form"
    >
      <wx-view
        class="weui-search-bar__box"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-input
          class="weui-search-bar__input"
          focus="{{true}}"
          placeholder="搜索"
          type="text"
          value="123"
          bind:blur="inputBlur"
          bind:focus="inputFocus"
          bind:input="inputChange"
        />
        <wx-text
          class="weui-icon-clear"
          bind:tap="clearInput"
        />
      </wx-view>
      <wx-label
        class="weui-search-bar__label"
        bind:tap="showInput"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-text
          class="weui-search-bar__text"
        >
          搜索
        </wx-text>
      </wx-label>
    </wx-view>
    <wx-view
      class="weui-search-bar__cancel-btn"
      bind:tap="hideInput"
    >
      取消
    </wx-view>
  </wx-view>
  <mp-cells>
    <wx-view
      class=" searchbar-result  weui-cells__group  "
    >
      <wx-view
        class="weui-cells weui-cells_after-title "
      >
        <mp-cell
          class="result"
          data-index="{{0}}"
          bind:tap="selectResult"
        >
          <wx-view
            class="weui-cell    "
            hoverClass="weui-cell_active"
            bind:tap="navigateTo"
          >
            <wx-view
              class="weui-cell__hd "
            />
            <wx-view
              class="weui-cell__bd weui-cell_primary"
            >
              <wx-view>
                搜索结果123
              </wx-view>
            </wx-view>
            <wx-view
              class="weui-cell__ft "
            />
          </wx-view>
        </mp-cell>
      </wx-view>
    </wx-view>
  </mp-cells>
</mp-searchbar>
`;

exports[`searchbar basic 5`] = `
<mp-searchbar
  id="searchbar"
  bind:selectresult="onSelectResult"
>
  <wx-view
    class="weui-search-bar  "
  >
    <wx-view
      class="weui-search-bar__form"
    >
      <wx-view
        class="weui-search-bar__box"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-input
          class="weui-search-bar__input"
          focus="{{true}}"
          placeholder="搜索"
          type="text"
          value="123"
          bind:blur="inputBlur"
          bind:focus="inputFocus"
          bind:input="inputChange"
        />
        <wx-text
          class="weui-icon-clear"
          bind:tap="clearInput"
        />
      </wx-view>
      <wx-label
        class="weui-search-bar__label"
        bind:tap="showInput"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-text
          class="weui-search-bar__text"
        >
          搜索
        </wx-text>
      </wx-label>
    </wx-view>
  </wx-view>
</mp-searchbar>
`;

exports[`searchbar basic 6`] = `
<mp-searchbar
  id="searchbar"
  bind:selectresult="onSelectResult"
>
  <wx-view
    class="weui-search-bar weui-search-bar_focusing "
  >
    <wx-view
      class="weui-search-bar__form"
    >
      <wx-view
        class="weui-search-bar__box"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-input
          class="weui-search-bar__input"
          focus="{{true}}"
          placeholder="搜索"
          type="text"
          value=""
          bind:blur="inputBlur"
          bind:focus="inputFocus"
          bind:input="inputChange"
        />
      </wx-view>
      <wx-label
        class="weui-search-bar__label"
        bind:tap="showInput"
      >
        <wx-icon
          class="weui-icon-search"
          size="12"
          type="search"
        />
        <wx-text
          class="weui-search-bar__text"
        >
          搜索
        </wx-text>
      </wx-label>
    </wx-view>
    <wx-view
      class="weui-search-bar__cancel-btn"
      bind:tap="hideInput"
    >
      取消
    </wx-view>
  </wx-view>
</mp-searchbar>
`;
